### PR TITLE
[D1] Write operation definition and pricing update

### PR DESCRIPTION
**Pull request by Oxyjun**
### Summary

1. Changed bullet pointed FAQs into H3s.
2. Added a note component to direct users to `meta object` or Cloudflare dashboard to accurately track usage.
3. Modified definition bullet 2 to specify that `UPDATE` and `DELETE` count as writes.
4. Added definition bullet 3 to specify that DDLs can be both read and write.

### Documentation checklist

- [x] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [ ] If a larger change - such as adding a new page- an issue has been opened in relation to any incorrect or out of date information that this PR fixes.
- [ ] Files which have changed name or location have been allocated [redirects](https://developers.cloudflare.com/pages/configuration/redirects/#per-file).

### CODE REVIEW

File: src/content/docs/d1/platform/pricing.mdx [hunk 1]
```diff
@@ -3,16 +3,15 @@ pcx_content_type: concept
...
 
-import { Render } from "~/components"
+import { Render } from "~/components";
 
 D1 bills based on:
 
-* **Usage**: Queries you issue against D1 will count as rows read, rows written, or both (for transactions or batches).
-* **Scale-to-zero**: You are not billed for hours or capacity units. If you are not issuing queries against your database, you are not billed for compute.
-* **Storage**: You are only billed for storage above the included [limits](/d1/platform/limits/) of your plan.
+- **Usage**: Queries you issue against D1 will count as rows read, rows written, or both (for transactions or batches).
```
Comment (rozenmd) [inline src/content/docs/d1/platform/pricing.mdx]:
```suggestion
- **Usage**: Queries you run against D1 will count as rows read, rows written, or both (for transactions or batches).
```

File: src/content/docs/d1/platform/pricing.mdx [hunk 2]
```diff
@@ -3,16 +3,15 @@ pcx_content_type: concept
...
+- **Scale-to-zero**: You are not billed for hours or capacity units. If you are not issuing queries against your database, you are not billed for compute.
```
Comment (rozenmd) [inline src/content/docs/d1/platform/pricing.mdx]:
```suggestion
- **Scale-to-zero**: You are not billed for hours or capacity units. If you are not running queries against your database, you are not billed for compute.
```

File: src/content/docs/d1/platform/pricing.mdx [hunk 3]
```diff
@@ -22,25 +21,25 @@ D1 bills based on:
...
 
-* Will D1 always have a Free plan?
+### Will D1 always have a Free plan?
 
 Yes, the [Workers Free plan](/workers/platform/pricing/#workers) will always include the ability to prototype and experiment with D1 for free.
 
-* What happens if I exceed the daily limits on reads and writes, or the total storage limit, on the Free plan?
+### What happens if I exceed the daily limits on reads and writes, or the total storage limit, on the Free plan?
 
 When your account hits the daily read and/or write limits, you will not be able to issue queries against D1. D1 API will return errors to your client indicating that your daily limits have been exceeded. Once you have reached your included storage limit, you will need to delete unused databases or clean up stale data before you can insert new data, create or alter tables or create indexes and triggers.
```
Comment (rozenmd) [inline src/content/docs/d1/platform/pricing.mdx]:
```suggestion
When your account hits the daily read and/or write limits, you will not be able to run queries against D1. D1 API will return errors to your client indicating that your daily limits have been exceeded. Once you have reached your included storage limit, you will need to delete unused databases or clean up stale data before you can insert new data, create or alter tables or create indexes and triggers.
```

**Review by rozenmd — COMMENTED** [review]
Can we replace "issue" with "run"?

as a developer, I've never heard of "issuing queries", only ever "running queries".

File: src/content/docs/d1/platform/pricing.mdx [hunk 4]
```diff
@@ -55,22 +54,22 @@ Every query returns a `meta` object that contains a total count of the rows read
...
 
-* Does D1 charge for data transfer / egress?
+### Does D1 charge for data transfer / egress?
 
 No.
 
-* Does D1 charge additional for additional compute?
+### Does D1 charge additional for additional compute?
 
 D1 itself does not charge for additional compute. Workers querying D1 and computing results: for example, serializing results into JSON and/or issuing queries, are billed per [Workers pricing](/workers/platform/pricing/#workers), in addition to your D1 specific usage.
```
Comment (rozenmd) [inline src/content/docs/d1/platform/pricing.mdx]:
```suggestion
D1 itself does not charge for additional compute. Workers querying D1 and computing results: for example, serializing results into JSON and/or running queries, are billed per [Workers pricing](/workers/platform/pricing/#workers), in addition to your D1 specific usage.
```

File: src/content/docs/d1/platform/pricing.mdx [hunk 5]
```diff
@@ -55,22 +54,22 @@ Every query returns a `meta` object that contains a total count of the rows read
...
 
-* Do queries I issue from the dashboard or wrangler (the CLI) count as billable usage?
+### Do queries I issue from the dashboard or Wrangler (the CLI) count as billable usage?
```
Comment (rozenmd) [inline src/content/docs/d1/platform/pricing.mdx]:
```suggestion
### Do queries I run from the dashboard or Wrangler (the CLI) count as billable usage?
```

File: src/content/docs/d1/platform/pricing.mdx [hunk 6]
```diff
@@ -55,22 +54,22 @@ Every query returns a `meta` object that contains a total count of the rows read
...
 
 Yes, any queries you issue against your database, including inserting (`INSERT`) existing data into a new database, table scans (`SELECT * FROM table`), or creating indexes count as either reads or writes.
```
Comment (rozenmd) [inline src/content/docs/d1/platform/pricing.mdx]:
```suggestion
Yes, any queries you run against your database, including inserting (`INSERT`) existing data into a new database, table scans (`SELECT * FROM table`), or creating indexes count as either reads or writes.
```

**Review by Maddy-Cloudflare — APPROVED** [review]

**Review by rozenmd — COMMENTED** [review]
I missed a few

File: src/content/partials/workers/d1-pricing.mdx [hunk 7]
```diff
@@ -1,22 +1,25 @@
...
-| Storage (per GB stored)         | 5 GB (total)                                        | First 5 GB included + $0.75 / GB-mo |
-
+|                         | [Workers Free](/workers/platform/pricing/#workers) | [Workers Paid](/workers/platform/pricing/#workers)        |
+| ----------------------- | -------------------------------------------------- | --------------------------------------------------------- |
+| Rows read               | 5 million / day                                    | First 25 billion / month included + $0.001 / million rows |
+| Rows written            | 100,000 / day                                      | First 50 million / month included + $1.00 / million rows  |
+| Storage (per GB stored) | 5 GB (total)                                       | First 5 GB included + $0.75 / GB-mo                       |
 
+:::note[Track your D1 usage]
+To accurately track your usage, use the [meta object](/d1/build-with-d1/d1-client-api/#return-object) or the [Cloudflare dashboard](https://dash.cloudflare.com/login). From your dashboard, navigate to: Storage & Databases > D1 > `[YOUR_DATABASE]` > Metrics > Row Metrics.
```
Comment (lambrospetrou) [inline src/content/partials/workers/d1-pricing.mdx]: Can we point the Cloudflare dashboard link to the D1 page directly?

Comment (Oxyjun) [reply src/content/partials/workers/d1-pricing.mdx]: I didn't realise it was possible, but it was. Have updated the link to land on Workers > D1, and have updated the instruction to reflect this change. Thanks!

File: src/content/partials/workers/d1-pricing.mdx [hunk 8]
```diff
@@ -1,22 +1,25 @@
...
-| Rows written                    | 100,000 / day                                      | First 50 million / month included + $1.00 / million rows |
-| Storage (per GB stored)         | 5 GB (total)                                        | First 5 GB included + $0.75 / GB-mo |
-
+|                         | [Workers Free](/workers/platform/pricing/#workers) | [Workers Paid](/workers/platform/pricing/#workers)        |
+| ----------------------- | -------------------------------------------------- | --------------------------------------------------------- |
+| Rows read               | 5 million / day                                    | First 25 billion / month included + $0.001 / million rows |
+| Rows written            | 100,000 / day                                      | First 50 million / month included + $1.00 / million rows  |
+| Storage (per GB stored) | 5 GB (total)                                       | First 5 GB included + $0.75 / GB-mo                       |
 
+:::note[Track your D1 usage]
```
Comment (harshil1712) [inline src/content/partials/workers/d1-pricing.mdx:11]: Can we also add the GraphQL Analytics API (/d1/observability/metrics-analytics/)

File: src/content/partials/workers/d1-pricing.mdx [hunk 9]
```diff
@@ -1,22 +1,25 @@
...
-6. Storage is based on gigabytes stored per month, and is based on the sum of all databases in your account. Tables and indexes both count towards storage consumed.
-7. Free limits reset daily at 00:00 UTC. Monthly included limits reset based on your monthly subscription renewal date, which is determined by the day you first subscribed.
-8. There are no data transfer (egress) or throughput (bandwidth) charges for data accessed from D1.
+2. Rows written measure how many rows were written to D1 database. Write operations include `INSERT`, `UPDATE`, and `DELETE`. Each of these operations contribute towards rows written. A query that `INSERT` 10 rows into a `users` table would count as 10 rows written.
+3. DDL operations (for example, `CREATE`, `ALTER`, and `DROP`) are used to define or modify the structure of a database. They may contribute to a mix of read rows and write rows. Ensure you are accurately tracking your usage through the available tools.
```
Comment (vy-ton) [inline src/content/partials/workers/d1-pricing.mdx:19]: `through the available tools` - should link to the FAQ on this

**Review by vy-ton — COMMENTED** [review]
We can delete the FAQ for `When will D1 start billing me?` - this was before D1 started billing

**Pull request merged**